### PR TITLE
ADD New package revocap-refiner

### PR DESCRIPTION
**Pull request by h-denpo**
"""The University of Tokyo, CISS Project:
        Geometric processing, mesh processing, mesh generation"""

### CODE REVIEW

File: var/spack/repos/builtin/packages/revocap-refiner/package.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+
+class RevocapRefiner(MakefilePackage):
+    """The University of Tokyo, CISS Project:
+        Geometric processing, mesh processing, mesh generation"""
+
+    homepage = "https://github.com/FrontISTR/FrontISTR"
+    git      = "https://github.com/FrontISTR/REVOCAP_Refiner.git"
+
+    version('master', commit='626d445')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/revocap-refiner/package.py]:
```suggestion
    version('master', branch='master')
```

File: var/spack/repos/builtin/packages/revocap-refiner/package.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    depends_on('ruby', type='build')
+    depends_on('mpi')
+    depends_on('doxygen', type='build')
+    depends_on('swig')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/revocap-refiner/package.py]:
```suggestion
    depends_on('swig', type='build')
```

File: var/spack/repos/builtin/packages/revocap-refiner/package.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        m.filter(r'RM\s*=.*$', 'RM=rm -f')
+        m.filter(r'DOXYGEN\s*=.*$', 'DOXYGEN=doxygen')
+        m.filter(r'TAR\s*=.*$', 'TAR=tar')
+        m.filter(r'SWIG\s*=.*$', 'SWIG=swig')
+
+    def install(self, spec, prefix):
+        install_tree('bin', prefix.bin)
+        install_tree('lib', prefix.lib)
+        mkdirp(prefix.include)
+        install_tree('Refiner', prefix.include)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/revocap-refiner/package.py]:
```suggestion
        install_tree('Refiner', prefix.include.Refine)
```

File: var/spack/repos/builtin/packages/revocap-refiner/package.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        m.filter(r'LIBPATH\s*=.*$', 'LIBPATH= ')
+        m.filter(r'RM\s*=.*$', 'RM=rm -f')
+        m.filter(r'DOXYGEN\s*=.*$', 'DOXYGEN=doxygen')
+        m.filter(r'TAR\s*=.*$', 'TAR=tar')
+        m.filter(r'SWIG\s*=.*$', 'SWIG=swig')
+
+    def install(self, spec, prefix):
+        install_tree('bin', prefix.bin)
+        install_tree('lib', prefix.lib)
+        mkdirp(prefix.include)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/revocap-refiner/package.py]: I don't think this line is actually needed

File: var/spack/repos/builtin/packages/revocap-refiner/package.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class RevocapRefiner(MakefilePackage):
+    """The University of Tokyo, CISS Project:
+        Geometric processing, mesh processing, mesh generation"""
+
+    homepage = "https://github.com/FrontISTR/FrontISTR"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/revocap-refiner/package.py]: Why doesn't the homepage match the same git repository?

Comment (h-denpo) [reply var/spack/repos/builtin/packages/revocap-refiner/package.py]: I made a mistake in writing.
The correct URL is https://github.com/FrontISTR/REVOCAP_Refiner

File: var/spack/repos/builtin/packages/revocap-refiner/package.py (hunk 4 above)
Comment (h-denpo) [reply var/spack/repos/builtin/packages/revocap-refiner/package.py]: I think so, too.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**